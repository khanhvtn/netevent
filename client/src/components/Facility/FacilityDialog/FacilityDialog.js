import React from 'react';
import {
    CircularProgress,
    Button,
    Select,
    MenuItem,
    Dialog,
    DialogTitle,
    DialogContent,
    DialogActions,
    TextField,
    Collapse,
    DialogContentText,
    Slide,
    InputLabel,
    FormControl,
} from '@material-ui/core';
import { Alert } from '@material-ui/lab';
import { useSelector } from 'react-redux';
//import useStyles in the last
import useStyles from './styles';
const Transition = React.forwardRef(function Transition(props, ref) {
    return <Slide direction="up" ref={ref} {...props} />;
});
const FacilityDialog = ({
    openCreateAndUpdateDialog,
    handleToggleDialogCreateAndUpdate,
    isCreateMode,
    openAlert,
    handleChange,
    name,
    code,
    type,
    openDeleteDialog,
    handleCreateAndUpdate,
    handleToggleDialogDelete,
    handleDelete,
}) => {
    const css = useStyles();
    const { isLoading, errors, createSuccess } = useSelector((state) => ({
        isLoading: state.facility.isLoading,
        errors: state.error.errors,
        createSuccess: state.facility.createSuccess,
    }));
    return (
        <div>
            {/* Dialog Create and Update */}
            <Dialog
                TransitionComponent={Transition}
                maxWidth="sm"
                open={openCreateAndUpdateDialog}
                fullWidth
                className={css.dialogCreateUpdateFac}
                onClose={(e) => handleToggleDialogCreateAndUpdate(e)}
                aria-labelledby="form-dialog-title"
            >
                <DialogTitle id="form-dialog-title">
                    {isCreateMode ? 'Create New Facility' : 'Update a Facility'}
                </DialogTitle>
                <DialogContent>
                    <Collapse in={openAlert} className={css.textField}>
                        <Alert severity="success">
                            Create New Facility Success!
                        </Alert>
                    </Collapse>
                    <TextField
                        disabled={isLoading || createSuccess ? true : false}
                        className={css.textField}
                        helperText={errors?.name ? errors?.name : ''}
                        error={errors?.name ? true : false}
                        variant="outlined"
                        onChange={handleChange}
                        id="name"
                        value={name}
                        name="name"
                        label="Name"
                        type="text"
                        fullWidth
                    />
                    <TextField
                        disabled={isLoading || createSuccess ? true : false}
                        className={css.textField}
                        helperText={errors?.code ? errors?.code : ''}
                        error={errors?.code ? true : false}
                        variant="outlined"
                        onChange={handleChange}
                        id="code"
                        value={code}
                        name="code"
                        label="Code"
                        type="text"
                        fullWidth
                    />
                    <TextField
                        disabled={isLoading || createSuccess ? true : false}
                        className={css.textField}
                        helperText={errors?.type ? errors?.type : ''}
                        error={errors?.type ? true : false}
                        variant="outlined"
                        onChange={handleChange}
                        id="type"
                        value={type}
                        name="type"
                        label="Type"
                        type="text"
                        fullWidth
                    />
                </DialogContent>
                <DialogActions className={css.dialogActions}>
                    <Button
                        disabled={isLoading || createSuccess ? true : false}
                        onClick={handleToggleDialogCreateAndUpdate}
                        color="default"
                    >
                        Cancel
                    </Button>
                    <Button
                        disabled={isLoading || createSuccess ? true : false}
                        variant="contained"
                        onClick={handleCreateAndUpdate}
                        color="primary"
                    >
                        {isLoading ? (
                            <CircularProgress size={25} color="inherit" />
                        ) : isCreateMode ? (
                            'Create'
                        ) : (
                            'Update'
                        )}
                    </Button>
                </DialogActions>
            </Dialog>

            {/* Dialog Confirm Delete */}
            <Dialog
                TransitionComponent={Transition}
                open={openDeleteDialog}
                onClose={handleToggleDialogDelete}
                aria-labelledby="delete-dialog"
                fullWidth
                className={css.dialogDeleteFac}
                aria-describedby="delete-dialog-description"
            >
                <DialogTitle id="delete-dialog">{'Warning!!!'}</DialogTitle>
                <DialogContent>
                    <DialogContentText id="delete-dialog-description">
                        Are you sure with your action ?
                    </DialogContentText>
                </DialogContent>
                <DialogActions className={css.dialogActions}>
                    <Button
                        disabled={isLoading ? true : false}
                        onClick={handleToggleDialogDelete}
                        color="default"
                    >
                        Cancel
                    </Button>
                    <Button
                        disabled={isLoading ? true : false}
                        variant="contained"
                        onClick={handleDelete}
                        color="secondary"
                    >
                        {isLoading ? (
                            <CircularProgress size={25} color="inherit" />
                        ) : (
                            'Delete'
                        )}
                    </Button>
                </DialogActions>
            </Dialog>
        </div>
    );
};

export default FacilityDialog;
