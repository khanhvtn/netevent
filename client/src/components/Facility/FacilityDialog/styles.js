import { makeStyles } from '@material-ui/core/styles';

export default makeStyles((theme) => ({
    textField: {
        marginBottom: '20px',
    },
    dialogActions: {
        marginRight: '20px',
    },

    dialogCreateUpdateFac: {
        minWidth: 500,
        [theme.breakpoints.down('sm')]: {
            minWidth: 0,
        }
    },

    dialogDeleteFac: {
        minWidth: 500,
        [theme.breakpoints.down('sm')]: {
            minWidth: 0,
        }
    }
}));
