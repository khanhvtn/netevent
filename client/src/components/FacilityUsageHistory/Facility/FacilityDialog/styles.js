import { makeStyles } from '@material-ui/core/styles';

export default makeStyles((theme) => ({
    textField: {
        marginBottom: '20px'
    },
    dialogActions: {
        marginRight: 16,
        marginBottom: 8,
        marginTop: 8
    },

    dialogCreateUpdateFac: {
        minWidth: 500,
        [theme.breakpoints.down('sm')]: {
            minWidth: 0
        }
    },

    dialogDeleteFac: {
        minWidth: 500,
        [theme.breakpoints.down('sm')]: {
            minWidth: 0
        }
    }
}));
