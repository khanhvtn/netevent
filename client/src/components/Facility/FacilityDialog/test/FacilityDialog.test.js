import React from 'react'
import { Provider } from 'react-redux'
import { renderer } from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import beforeEach from 'jest-wrap'
import expect from 'expect'
import toJson from 'enzyme-to-json'

import FacilityDialog from '../FacilityDialog'

const mockStore = configureStore([])
describe('FacilityDialog', () => {
    let store;
    let component;

    beforeEach(() => {
        store = mockStore({
            myState: 'Facility Dialog'
        })

        store.dispatch = jest.fn()

        component = renderer.create(
            <Provider store={store}>
                <FacilityDialog />
            </Provider>
        )
    })
    
    it('should render with given state from Redux store', () => {
        expect(toJson(component)).toMatchSnapshot()
    })
})