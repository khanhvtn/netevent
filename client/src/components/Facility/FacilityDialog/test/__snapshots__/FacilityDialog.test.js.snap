// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacilityDialog should render with given state from Redux store 1`] = `undefined`;
